hikari/guilds.py

```diff
@@ -487,6 +498,15 @@ def mention(self) -> str:
         """
         return f"<@!{self.id}>" if self.nickname is not None else self.user.mention
 
+    def is_timed_out(self) -> bool:
```

Comment:
Why not have a "raw_communication_disabled_until" field with a communication_disabled_until property where the property only returns the datetime if its not in the past so that you don't have to use both fields or just do the check for whether its in the past or not yourself to work out when their timeout expires?

Reply:
Thommo and I had a discussion about this, the problem with the property is that people may rely on it and during the execution of the code, it might return None instead of the object you want. It is unlikely to happen, but it can.

Ie:

```py
if member.communication_disabled_until is not None:
    # Do something with it, but once its accessed again it may be None
```

Having a function call clearly denotes that it might change the next time it's called as well as provide a more helpful function

Reply:
Then why can't communication_disabled_until be a function

Reply:
The reason I didnt like it is because it breaks a bit the standard if suddenly the way of accessing that attribute is through a function call. 

Would also like to get @tandemdude's opinion on this so we can find the best way to approach this issue

Reply:
Personally I think I prefer the `is_x` style as it is more consistent with the other functions

Reply:
> The reason I didnt like it is because it breaks a bit the standard if suddenly the way of accessing that attribute is through a function call.

is_timed_out also has the name (and doc string funnily enough) of an attribute so

Reply:
Actually thinking about it more I think I'm with @FasterSpeeding here it would probably make more sense to be a function retuning a datetime or None given the that other `is_x` members are actually properties

Reply:
Ok, then I guess we can change it to a function for it :+1: